drivers/gpio/gpio_sx1509b.c

```diff
@@ -320,12 +460,55 @@ static int pin_interrupt_configure(struct device *dev,
 				   enum gpio_int_mode mode,
 				   enum gpio_int_trig trig)
 {
-	int ret = 0;
+	const struct sx1509b_config *cfg = dev->config->config_info;
+	struct sx1509b_drv_data *drv_data = dev->driver_data;
+	struct sx1509b_pin_state *pins = &drv_data->pin_state;
+	int rc = 0;
 
-	if (mode != GPIO_INT_MODE_DISABLED) {
-		ret = -ENOTSUP;
+	if ((mode != GPIO_INT_MODE_DISABLED) &&
```

Comment:
This should be:

	if (!IS_ENABLED(CONFIG_GPIO_EX1509B_INTERRUPT)
	    && (mode != GPIO_INT_MODE_DISABLED)) {
		return -ENOTSUP;
	}

Reply:
Done!